woocommerce/woocommerce-gutenberg-products-block
# Honor hidden property of cart item data and add support for experimenal property

This PR makes it so:
* If a cart item data has the `hidden` property set to true, it will be hidden.
* `hidden` can be overridden with the `__experimental_woocommerce_blocks_hidden` property, which only has effect in blocks.

### Screenshots
![imatge](https://user-images.githubusercontent.com/3616980/105713722-bfb50780-5f1b-11eb-87f3-a910a5fc2251.png)

![imatge](https://user-images.githubusercontent.com/3616980/105713740-c5aae880-5f1b-11eb-82b2-ee3fe779337d.png)

### How to test the changes in this Pull Request:

0. Checkout `fix/details-disappearing` branch in WC Subscriptions (see relevant PR: 3968-gh-woocommerce/woocommerce-subscriptions).
1. Add a subscription with free trial and signup fee to your cart.
2. Go to the Cart block and verify the free trial and signup fee details are rendered.
4. Open the Cart shortcode and verify free trial and signup fee metadata is not rendered.

### Changelog

> Hidden cart item meta data will not be rendered in the Cart and Checkout blocks.
